poweriq-worker/src/main/java/com/vmware/flowgate/poweriqworker/jobs/PowerIQService.java

```diff
@@ -894,80 +888,153 @@ public PowerIQAPIClient createClient(FacilitySoftwareConfig poweriq) {
       return realTimeDatas;
    }
 
-   public List<ValueUnit> getValueUnits(Pdu pdu,
+   public List<ValueUnit> getValueUnits(String pduIDFromPowerIQ, PowerIQAPIClient client,
          HashMap<AdvanceSettingType, String> advanceSettingMap) {
       List<ValueUnit> values = new ArrayList<ValueUnit>();
-      if (pdu == null || pdu.getReading() == null || pdu.getReading().getInletReadings() == null) {
+      Long pduId = Long.parseLong(pduIDFromPowerIQ);
+      List<Outlet> outlets = client.getOutlets(pduId);
```

Comment:
Can we got those reading from pdu directly?

Reply:
beside that where is the sensor data ?  We should also query the sensor data from the powerIQ in syncrealtime data job.

Reply:
1>We only can read the inlet reading info from pdu directly, Actually, we don't need to get the pdu first. We can get the outlets and inlets reading info directly by PduID. 
2>We have an another job to sync realtime data for sensor